packages/driver/cypress/e2e/commands/cookies.cy.js

```diff
@@ -9,148 +9,142 @@ describe('src/cy/commands/cookies - no stub', () => {
     // this can be removed along with the experimental flag since once the flag
     // removed, clearing cookies for all domains will be done by default
     beforeEach(() => {
-      if (!Cypress.config('experimentalSessionAndOrigin')) {
-        cy.clearCookies({ domain: null })
-      }
+      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
+
+      cy.setCookie('key1', 'value1', { domain: 'www.foobar.com' })
+      cy.setCookie('key2', 'value2', { domain: 'foobar.com' })
+      cy.setCookie('key3', 'value3') // defaults to hostname: www.barbaz.com
+      cy.setCookie('key4', 'value4', { domain: 'www.barbaz.com' })
+      cy.setCookie('key5', 'value5', { domain: '.www.barbaz.com' })
+      cy.setCookie('key6', 'value6', { domain: 'barbaz.com' })
     })
 
     it('returns cookies from the domain matching the AUT by default', () => {
-      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-      cy.setCookie('baz', 'qux', { domain: 'foobar.com' })
-      cy.setCookie('foo', 'bar') // defaults to (super)domain: barbaz.com
-      cy.setCookie('qux', 'quuz', { domain: 'www.barbaz.com' })
-
       cy.getCookies().then((cookies) => {
-        expect(cookies).to.have.length(2)
-        // both the barbaz.com and www.barbaz.com cookies are yielded
-        expect(cookies[0].domain).to.match(/\.?barbaz\.com/)
-        expect(cookies[1].domain).to.match(/\.?www\.barbaz\.com/)
+        expect(cookies).to.have.length(3)
+
+        // order is not consistent between browsers or even guaranteed between runs
+        const sortedCookies = Cypress._.sortBy(cookies, 'name')
+
+        expect(sortedCookies[0].name).to.equal('key3')
+        expect(sortedCookies[0].domain).to.match(/\.?www\.barbaz\.com/)
+        expect(sortedCookies[1].name).to.equal('key4')
+        expect(sortedCookies[1].domain).to.match(/\.?www\.barbaz\.com/)
+        expect(sortedCookies[2].name).to.equal('key5')
+        expect(sortedCookies[2].domain).to.match(/\.?www\.barbaz\.com/)
```

Comment:
I would expect higher level domain cookies to be present here as well since the domain is `www.barbaz.com` unless the domain needs to be `.barbaz.com`?
```suggestion
        expect(sortedCookies[2].domain).to.match(/\.?www\.barbaz\.com/)
        expect(sortedCookies[3].name).to.equal('key6')
        expect(sortedCookies[3].domain).to.match(/\.?barbaz\.com/)
```